### ISSUE TITLE
train controlnet with img2img methods

### ISSUE THREAD

**Issue by VV-A-VV** [huggingface/diffusers#4421]
I want to train the controlnet with specific input of images and contol condition images instead of the prompts, but I found that the script only supports the prompts input to train the controlnet. 

Is any script which I didn't find can deal with this problem?

**Comment by sayakpaul** [comment]
The script supports images as well. Check out this block:

https://github.com/huggingface/diffusers/blob/main/examples/controlnet/train_controlnet.py#L1015C1-L1032C25

**Comment by VV-A-VV** [comment]
> The script supports images as well. Check out this block:
> 
> https://github.com/huggingface/diffusers/blob/main/examples/controlnet/train_controlnet.py#L1015C1-L1032C25

Let me rephrase my question, I am trying to be able to make the input an image during training, the controlnet's input also an image, and the output/target's image is not the same as the input's image. I noticed that this script's code its input image and target image are the same, which is not the same as what I was aiming for, as I wanted controlnet to learn specific conditions. Perhaps I could try rewriting the code slightly to achieve my goal?

**Comment by sayakpaul** [comment]
Could you provide a visual example of what you're asking for?

**Comment by VV-A-VV** [comment]
/
![IMG_1859](https://github.com/huggingface/diffusers/assets/59974890/30f86600-4f5f-49ec-bfb4-661585b841c5)
Sure, like the graph above, the dataset should contain three types of the images and the input image1 is different from the target image.

**Comment by sayakpaul** [comment]
No, this is not something we support. We haven't seen any community interest for this as well.

**Comment by VV-A-VV** [comment]
All right, thanks for your help. I will try to implement it by myself.

**Comment by wanglaofei** [comment]
The official controlnet training code works for text2img, how to modify the controlnet training code for img2img? I don't know if you have solved it, I would like to ask you. Thanks a lot!

**Comment by CuttlefishXuan** [comment]
@VV-A-VV Did you succeed in training controlnet with img2img stable diffusion? I'm also confused about this

**Comment by zazaza1234567** [comment]
> 官方的controlnet训练代码适用于text2img，如何修改img2img的controlnet训练代码？我不知道你有没有解决，我想问你。多谢！
 Did you succeed in training controlnet with img2img stable diffusion? I'm also confused about this
